File: BE-develop/fundyou1/src/main/java/com/ssafy/fundyou1/fund/dto/FundingItemResponseDto.java
package com.ssafy.fundyou1.fund.dto;

import com.ssafy.fundyou1.fund.entity.FundingItem;
import com.ssafy.fundyou1.item.dto.ItemResponseDto;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class FundingItemResponseDto {
    private Long fundingItemId;
    private ItemResponseDto item;
    private Long fundingId;
    // 참여자 수
    private int attendMemberCount;

    // 총가격
    private int itemTotalPrice;

    private int count;

    // 현재 펀딩된 가격
    private int currentFundingPrice;

    private boolean fundingItemStatus;

    @Builder
    public static FundingItemResponseDto createFundingItemDto(FundingItem fundingItem, int attendMemberCount) {
        return new FundingItemResponseDto(
                fundingItem.getId(),
                new ItemResponseDto(fundingItem.getItem(), false),
                fundingItem.getFunding().getId(),
                attendMemberCount,
                fundingItem.getItemTotalPrice(),
                fundingItem.getCount(),
                fundingItem.getCurrentFundingPrice(),
                fundingItem.isFundingItemStatus()
        );
    }


}

File: Android/FundYou/data/src/main/java/com/ssafy/fundyou/data/remote/datasource/funding/dto/FundingItemResponseDto.kt
package com.ssafy.fundyou.data.remote.datasource.funding.dto

import com.google.gson.annotations.SerializedName
import com.ssafy.fundyou.data.remote.datasource.item.dto.ItemResponseDto

internal data class FundingItemResponseDto(
    @SerializedName("count")
    val count: Int?,
    @SerializedName("currentFundingPrice")
    val currentFundingPrice: Int?,
    @SerializedName("hostName")
    val fundingHostUserName : String?,
    @SerializedName("fundingItemStatus")
    val fundingItemStatus: Boolean?,
    @SerializedName("fundingId")
    val fundingId: Long,
    @SerializedName("fundingItemId")
    val id: Long?,
    @SerializedName("arImgList")
    var arImgList: List<String>?,
    @SerializedName("attendMemberCount")
    val participateCount: Int?,
    @SerializedName("item")
    val itemInfo: ItemResponseDto
)